lua/oil/util.lua

```diff
@@ -360,28 +360,38 @@ M.is_floating_win = function(winid)
   return vim.api.nvim_win_get_config(winid or 0).relative ~= ""
 end
 
+---Recalculate the window title for the current buffer
+---@param winid nil|integer
+---@return string
+M.get_title = function(winid)
+  local src_buf = vim.api.nvim_win_get_buf(winid or 0)
+  local title = vim.api.nvim_buf_get_name(src_buf)
+  local scheme, path = M.parse_url(title)
+
+  if config.adapters[scheme] == "files" then
+    assert(path)
+    local fs = require("oil.fs")
+
+    if config.float.get_win_title ~= nil then
+      return config.float.get_win_title(fs.posix_to_os_path(path))
+    end
```

Comment:
I think that to make this more general, the config `get_win_title` function should take a winid and return a string. Then we could move this check-return to the top of the function.

Reply:
Thank you for the response. Implemented the feedback.